kube-derive/src/custom_resource.rs

```diff
@@ -1,496 +1,324 @@
-use crate::{CustomDerive, ResultExt};
+use darling::FromDeriveInput;
 use inflector::string::pluralize::to_plural;
 use proc_macro2::{Ident, Span};
-use syn::{Data, DeriveInput, Path, Result, Visibility};
-
-#[derive(Debug)]
-pub(crate) struct CustomResource {
-    tokens: proc_macro2::TokenStream,
-    ident: proc_macro2::Ident,
-    visibility: Visibility,
-    kubeattrs: KubeAttrs,
-}
+use syn::{DeriveInput, Path, Result};
 
 /// Values we can parse from #[kube(attrs)]
-#[derive(Debug, Default)]
-struct KubeAttrs {
+#[derive(Debug, Default, FromDeriveInput)]
+#[darling(attributes(kube))]
+pub(crate) struct KubeAttrs {
     group: String,
     version: String,
     kind: String,
-    kind_struct: String,
+    #[darling(default, rename = "struct")]
+    kind_struct: Option<String>,
     /// lowercase plural of kind (inferred if omitted)
+    #[darling(default)]
     plural: Option<String>,
+    #[darling(default)]
     namespaced: bool,
+    #[darling(default = "default_apiext")]
     apiextensions: String,
+    #[darling(multiple, rename = "derive")]
     derives: Vec<String>,
+    #[darling(default)]
     status: Option<String>,
+    #[darling(multiple, rename = "shortname")]
     shortnames: Vec<String>,
```

Comment:
You'll get a nice error like this:
![image](https://user-images.githubusercontent.com/639336/108990503-d00bef80-764b-11eb-95c6-b94fbf1f7066.png)

Reply:
Oo, that's nice. Assume this is using rust-analyzer?

Reply:
I'm using rust-analyzer, but "did you mean" is `darling`'s feature https://github.com/TedDriggs/darling/blob/d128844d71445bef93c6449612d16d6dc28625e8/core/src/error/kind.rs#L175

Reply:
Yeah, it was more that RA sometimes has issues with certain derive macros for me. But if this works fine then that's great.